### PR TITLE
Leaflet Map and Data Points for Demo

### PR DESCRIPTION
**Pull request by jedjas12**
Stripped down commits that contain only confirmed working and non-breaking changes for the Leaflet map in order to merge with Dev for the demo.

### CODE REVIEW

File: db/api/v1/controllers/__tests__/geometrySearch.controller.test.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    });
+
+    it('catches database access exceptions to pass them to the error handler', async () => {
+        const DB_ERROR = new Error();
+        geomDb.polygonFind = jest.fn(async () => { throw DB_ERROR; });
+        await polygonFind(req, res, next);
+        expect(next).toHaveBeenCalledWith(DB_ERROR);
+        expect(res.json).not.toHaveBeenCalled();
+    });
+});
```
Comment (APalumbo97) [inline db/api/v1/controllers/__tests__/geometrySearch.controller.test.js]: Newline at the end of the file.

File: db/api/v1/controllers/geometrySearch.controller.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        next(error);
+    }
+};
+
+module.exports = {
+    distanceFind,
+    envelopeFind,
+    polygonFind
+};
+
```
Comment (APalumbo97) [inline db/api/v1/controllers/geometrySearch.controller.js]: One of these newlines can be removed

File: db/api/v1/db/__tests__/geometrySearch.db.test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        // Create the linestring
+        const lineString = makeLineString(polygonPoints);
+        
+        await polygonFind(points);
+        expect(query).toHaveBeenCalledTimes(1);
+        expect(query.mock.calls[0][0]).toEqual(expect.stringContaining('ST_Within(a.location, ST_MakePolygon(ST_GeomFromText($1)))'));
+        expect(query.mock.calls[0][1][0]).toEqual(expect.stringContaining(lineString));
+    });
+});
```
Comment (APalumbo97) [inline db/api/v1/db/__tests__/geometrySearch.db.test.js]: Newline at the end of the file

File: db/api/v1/db/assets.db.js [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
-const QUERY_FIND = `SELECT
-sponsor.name AS sponsor_name,
-project.name AS project_name,
-asset.id AS asset_id,
-asset_type.name AS asset_type,
-asset_type.description AS asset_description,
-ST_X(asset.location) AS latitude,
-ST_Y(asset.location) AS longitude
+const QUERY_FIND = `
+SELECT
```
Comment (APalumbo97) [inline db/api/v1/db/assets.db.js]: Can this be tabbed over once so it is clear it is a part of QUERY_FIND?

File: db/api/v1/routes/validate.js [hunk 5]
```diff
@@ -118,6 +118,64 @@ const extractBodyParam = (req, paramName) => req.body[paramName];
...
+
+/**
+ * Parses a list of latitude and longitude points.
+ * @param {Array} coordinateList - a list of latitude and longitude points.
+ * @returns {ParseResult} parse success if a valid list of latitude and longitude points of at least length 3
+ */
+const parseCoordinates = (coordinateList) => {
+   
+    // There needs to be at least 3 
+    if (coordinateList.length < 3)
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Add braces around this code block

File: db/api/v1/routes/validate.js [hunk 6]
```diff
@@ -118,6 +118,64 @@ const extractBodyParam = (req, paramName) => req.body[paramName];
...
+        return ParseResult.failure('Expected at least 3 points in the coordinates list');
+
+    let lat = 0;
+    let lon = 0;
+
+    const coordinates = [];
+
+    var i;
+    var point;
+    for(i = 0; i < coordinateList.length; i++)
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Space after the `for`, also move the `{` up to the previous line

File: db/api/v1/routes/validate.js [hunk 7]
```diff
@@ -118,6 +118,64 @@ const extractBodyParam = (req, paramName) => req.body[paramName];
...
+    {
+        point = coordinateList[i];
+        lon = parseFloat(point.longitude);
+        lat = parseFloat(point.latitude);
+
+        if(!isNaN(lon) && validLongitude(lon)
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Space after `if`, also move `{` up a line

File: db/api/v1/routes/validate.js [hunk 8]
```diff
@@ -118,6 +118,64 @@ const extractBodyParam = (req, paramName) => req.body[paramName];
...
+        point = coordinateList[i];
+        lon = parseFloat(point.longitude);
+        lat = parseFloat(point.latitude);
+
+        if(!isNaN(lon) && validLongitude(lon) 
+            && !isNaN(lat) && validLatitude(lat))
+        {
+            coordinates.push({latitude: lat, longitude: lon});
+        }
+        else
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Same as if

File: db/api/v1/routes/validate.js [hunk 9]
```diff
@@ -131,6 +189,36 @@ const parseId = (idStr) => {
...
+ * Parse a latitude value from a string
+ * 
+ * @param {string} latStr - A string that is parsable into a floating point number
+ * @returns {ParseResult} - Parse success with a float value that is a latitude, or a failure message
+ */
+const parseLatitude = (latStr) => {
+    const lat = parseFloat(latStr);
+    const isValid = (!isNaN(lat) && validLatitude(lat));
+
+    return isValid ?
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Ternary operator noice

File: db/api/v1/utils/db.utils.js [hunk 10]
```diff
@@ -47,5 +76,6 @@ module.exports = {
     isValidDbInteger,
     beginTransaction,
     commitTransaction,
-    rollbackTransaction
+    rollbackTransaction,
+    makeLineString
 };
```
Comment (APalumbo97) [inline db/api/v1/utils/db.utils.js]: Add a newline to this file

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.html [hunk 11]
```diff
@@ -0,0 +1,7 @@
+<template>
+    <lightning-card title="Project Assets">
+        <div class="slds-m-around_medium">
+            <c-map onready={onMapInitialized}></c-map>
+        </div>
+    </lightning-card>
+</template>
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.html]: Newline

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 12]
```diff
@@ -0,0 +1,149 @@
+/* eslint-disable no-console */
+import { LightningElement } from 'lwc';
+
+import utils from 'c/utils';
+
+const DISTANCE_URL = utils.URL + 'assets/geometrySearch/distance';
+
+
+// import { assets, bboxAssets } from 'c/controllers';
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: Add a TODO for the commented out code so we don't forget about it

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+
+        
+        // Implement map "on click" handler
+        // If we could keep a reference to the map via a class variable (this.map = event.detail) this would be a lot cleaner
+        // to bad this map doesn't wish to cooperate at the time being.
+        map.on('click', function(e) {
+
+            // Remove the old radius search indicator so we can start a new query
+            if (this.mapCircle !== undefined)
+            {
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: Move `{` up a line

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+                map.removeLayer(this.mapCircle);
+            }
+
+            // Remove the old markers from the previous query
+            if (this.myAssets !== undefined && this.myAssets.length > 0)
+            {
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: `{`

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+                let i;
+                for(i = 0; i < this.myAssets.length; i++)
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: Space and `{`

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+            // Send out the GET request
+            this.assetsPromise = utils.get(distanceURL.href);
+
+            // When the promise is fulfilled handle it
+            this.assetsPromise.then(response => {
+                // Response should be an array of assets
+                console.log(response);
+
+                // Initialize the array if need be
+                if( this.myAssets === undefined){
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: Space after `if` and before `{`

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+                    this.myAssets = [];
+                }
+
+                // Iterate over the assets and add them to the map
+                let i;
+                for(i = 0; i < response.length; i++)
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: space and `{`

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+                    m.bindPopup(`${a.asset_type}\n\r${a.project_name}`);
+
+                    // Keep a reference to the marker so we can remove it later
+                    this.myAssets.push(m);
+                }
+            });
+        });
+        // this.map.on('click', this.onMapClicked);
+    }
+}
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: Newline

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js-meta.xml [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+    <masterLabel>Map - Geo Query</masterLabel>
+    <description>Query Assets by Location</description>
+    <targets>
+        <target>lightning__AppPage</target>
+        <target>lightning__RecordPage</target>
+        <target>lightning__HomePage</target>
+        <target>lightningCommunity__Page</target>
+        <target>lightningCommunity__Default</target>
+    </targets>
+</LightningComponentBundle>
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js-meta.xml]: Newline

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js [hunk 20]
```diff
@@ -26,14 +29,20 @@ export default class Map extends LightningElement {
         });
     }
 
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js]: This newline can be removed

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js [hunk 21]
```diff
@@ -26,14 +29,20 @@ export default class Map extends LightningElement {
...
      * Constructs the Leaflet map on the page and initializes this.map
      */
     initializeLeaflet() {
         const mapRoot = this.template.querySelector('.map-root');
         this.map = L.map(mapRoot);
         this.map.fitWorld();
+        L.control.scale().addTo(this.map);
+        this.map.setMinZoom(2);
+        this.map.setMaxZoom(17);
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js]: Extra newline can be removed

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js [hunk 22]
```diff
@@ -26,14 +29,20 @@ export default class Map extends LightningElement {
...
     }
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js]: Same here

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js [hunk 23]
```diff
@@ -45,7 +54,7 @@ export default class Map extends LightningElement {
             'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
             {
                 attribution: '&copy; ' + mapLink + ' Contributors',
-                maxZoom: 18
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js]: If the maxZoom is permanently gone, remove this newline

File: salesforce/my-conservation-life/force-app/main/default/lwc/utils/connection.js [hunk 24]
```diff
@@ -1,3 +1,3 @@
-const URL = 'https://my-conservation-life.herokuapp.com/api/v1/';
+const URL = 'https://my-conservation-life-dev4.herokuapp.com/api/v1/';
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/utils/connection.js]: Change this back to the original, this should not be checked into the repo

File: salesforce/my-conservation-life/force-app/main/default/staticresources/leafletMarkerCluster/leafletMarerClusterDefault.css [hunk 25]
```diff
@@ -0,0 +1,60 @@
+.marker-cluster-small {
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/staticresources/leafletMarkerCluster/leafletMarerClusterDefault.css]: This file could use some newlines between classes and the closing `}` moved one tab to the left for styling

File: salesforce/my-conservation-life/force-app/main/default/staticresources/leafletMarkerCluster/leafletMarerClusterDefault.css [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+	margin-left: 5px;
+	margin-top: 5px;
+
+	text-align: center;
+	border-radius: 15px;
+	font: 12px "Helvetica Neue", Arial, Helvetica, sans-serif;
+	}
+.marker-cluster span {
+	line-height: 30px;
+	}
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/staticresources/leafletMarkerCluster/leafletMarerClusterDefault.css]: Newline

**Comment by jedjas12** [comment]
@APalumbo97 @exw4141 Ready for review

File: db/api/v1/routes/validate.js [hunk 27]
```diff
@@ -118,6 +118,62 @@ const extractBodyParam = (req, paramName) => req.body[paramName];
...
+    for (i = 0; i < coordinateList.length; i++) {
+        point = coordinateList[i];
+        lon = parseFloat(point.longitude);
+        lat = parseFloat(point.latitude);
+
+        if (!isNaN(lon) && validLongitude(lon) 
+            && !isNaN(lat) && validLatitude(lat)) {
+            coordinates.push({latitude: lat, longitude: lon});
+        }
+        else {
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Move this else onto the previous line, https://www.w3schools.com/js/js_conventions.asp

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js [hunk 28]
```diff
@@ -15,6 +16,8 @@ export default class Map extends LightningElement {
      * 
      * When this is complete, call initializeleaflet()
      */
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js]: Remove these newlines

**Comment by jedjas12** [comment]
@APalumbo97

File: salesforce/my-conservation-life/force-app/main/default/staticresources/leafletMarkerCluster/leafletMarerClusterDefault.css (hunk 25 above)
Comment (exw4141) [reply salesforce/my-conservation-life/force-app/main/default/staticresources/leafletMarkerCluster/leafletMarerClusterDefault.css]: I strongly agree with the closing braces moving to the left

**Comment by APalumbo97** [comment]
@jedjas12 assuming all the comments have been resolved this should be ready to merge in now

**Comment by APalumbo97** [comment]
@jedjas12 Also make sure to resolve the merge conflicts

File: db/api/v1/routes/validate.js [hunk 29]
```diff
@@ -174,6 +174,134 @@ const parseCoordinates = (coordinateList) => {
...
+        }
+
+        lon = parseFloat(coordArray[0]);
+        lat = parseFloat(coordArray[1]);
+
+        if (!isNaN(lon) && validLongitude(lon)
+            && !isNaN(lat) && validLatitude(lat)) {
+            parsedCoordinates.push([lon, lat]);
+        }
+        else {
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: `else` can be moved up a line

Comment (APalumbo97) [reply db/api/v1/routes/validate.js]: Nvm too late

**Pull request merged**